mivideo/mivideo_extract.py

```diff
@@ -297,31 +296,45 @@ def _makeCourseData(resultDictionaries: Sequence[Dict], categoryFilter: str) ->
 
         :param resultDictionaries: a sequence of KalturaMediaEntry objects converted to dictionaries
-        :param categoryFilter: the base Kaltura category, usually `Canvas_UMich`
         :return:
         """
         courseData: pd.DataFrame = pd.DataFrame.from_records(
             resultDictionaries, columns=('id', 'categories',)
-        ).rename(columns={'id': 'media_id', 'categories': 'course_id', })
+        ).rename(columns={'id': 'media_id', })
 
-        courseData = courseData.assign(
-            course_id=courseData['course_id'].str.split(',')).explode('course_id')
+        # split category CSVs to multiple new rows
+        courseData = (
+            courseData
+                .assign(categories=courseData['categories'].map(lambda c: c.split(',')))
+                .explode('categories')
+        )
 
-        courseData['in_context'] = [
-            c.endswith('>InContext') for c in courseData['course_id']]
+        # remove rows whose category isn't in "Canvas_UMich>site>channels>" hierarchy
+        courseData = courseData[courseData['categories']
+            .map(lambda c: c.startswith('Canvas_UMich>site>channels>'))]
 
-        courseData['course_id'] = courseData['course_id'].str.replace(
-            r'^' + categoryFilter + r'.*>([0-9]+)(>InContext)?$',
-            lambda m: m.groups()[0], regex=True
-        )
+        courseData['in_context'] = courseData['categories'] \
+            .map(lambda c: c.endswith('>InContext'))
+
+        # get course ID from '>'-delimited category hierarchy string
+        columnIndex = 3
+        courseData['course_id'] = \
+            courseData['categories'].map(lambda c: (c + '>' * columnIndex).split('>')[columnIndex])
+
+        # categories have been processed, drop unneeded column
+        courseData = courseData.drop('categories', axis=1)  # 1: columns
+
+        # find and drop invalid, non-decimal course IDs
+        # (e.g., "Shared Repository", "5430bafe907cca901a0f11646470dd64244ebd5f", etc.)
+        validCourseIdIndex = courseData['course_id'].map(str.isdecimal)
 
-        # find and drop non-decimal course IDs
-        # (e.g., like category "Canvas_UMich>site>channels>Shared Repository")
-        badCourseIds = courseData[courseData['course_id'].str.contains('[^0-9]', regex=True)]
-        if (not badCourseIds.empty):
-            logger.debug(f'Non-numeric course IDs to be removed:\n{badCourseIds}')
+        if (not validCourseIdIndex.all()):  # not all all index items are True
+            invalidCourseIdIndex = ~validCourseIdIndex
+            logger.info(f'Removing ({invalidCourseIdIndex.values.sum()}) non-decimal course IDs...')
+            logger.debug(f'Course IDs to be removed:\n{courseData[invalidCourseIdIndex]}')
+            courseData = courseData[validCourseIdIndex]  # keep the valid ones
 
-        courseData = courseData[courseData['course_id'].str.isdecimal()].drop_duplicates()
+        courseData.drop_duplicates(inplace=True)
```

Comment:
Nit-pick: I'm personally not a fan of the mutating version of `drop_duplicates` (with `inplace`). Seems like things are moving toward deprecating this kind of usage: https://github.com/pandas-dev/pandas/issues/16529 

I actually had never used `drop_duplicates` in this way and didn't know about it until I was confused when looking at @jonespm's code on this project: see the closed issue #52. (GitHub issue discovery courtesy of @jonespm).

Reply:
Ah, too bad.  One of the things I don't like about Pandas is that the code tends to be too wordy with repetitions of the names of the Dataframes and Series.  I wish more functions had the option of using `inplace=`.

With the deprecation looking, I will change this code.